elephant/test/test_spectral.py

```diff
@@ -210,9 +211,9 @@ def test_welch_cohere_behavior(self):
         freqs_np, coherency_np, phase_lag_np =\
             elephant.spectral.welch_cohere(x.magnitude.flatten(), y.magnitude.flatten(),
                 fs=1/sampling_period, freq_res=freq_res)
-        self.assertTrue((freqs == freqs_np).all() and
-                        (coherency[:, 0] == coherency_np).all() and
-                        (phase_lag[:, 0] == phase_lag_np).all())
+        assert_array_almost_equal(freqs.simplified.magnitude, freqs_np)
+        assert_array_almost_equal(coherency[:, 0], coherency_np)
+        assert_array_almost_equal(phase_lag[:, 0], phase_lag_np)
```

Comment:
These assertions pass with the original test (using assertTrue) in my configuration.
Do they fail in other configurations?
If they do, what configurations?

Reply:
All numpy floating-point value arrays must be checked with the ground truth arrays by `assert_array_almost_equal()`. This is the numpy's testing convention and it simplifies the debugging because when the test fails it shows useful information about how many values differ, max absolute and relative difference instead of just
got False, expected True
message

Reply:
They failed once I started debugging and that's why I changed them to ones how it should be

Reply:
Then can't we use assert_array_equal?
These arrays should be exactly the same, as the computation inside the function should not depend on how the data were given.

Reply:
For example,
```python
    phase_lag_new = np.angle(Pxy.conj())

    from numpy.testing import assert_array_almost_equal
    assert_array_almost_equal(phase_lag, phase_lag_new)  # phase_lag is our old _welch()
```

```
AssertionError: 
Arrays are not almost equal to 6 decimals

Mismatched elements: 2 / 1000 (0.2%)
Max absolute difference: 6.28318531
Max relative difference: 2.
 x: array([[ 3.141593, -2.602966,  2.297019,  2.649172, -1.171436, -0.943932,
        -0.565363, -1.253926,  2.561917, -0.813005, -0.940448, -1.601776,
         2.712933,  2.158754,  0.957191,  1.645848, -0.283867, -1.849341,...
 y: array([[-3.141593, -2.602966,  2.297019,  2.649172, -1.171436, -0.943932,
        -0.565363, -1.253926,  2.561917, -0.813005, -0.940448, -1.601776,
         2.712933,  2.158754,  0.957191,  1.645848, -0.283867, -1.849341,...
```

this is actually how it behaves now - it does not much precisely even with `.conj()`

Reply:
Please use also here assert_array_equal().

Reply:
done